### DISCUSSION TITLE
Handle Undefined Attributes with TypeScript

### DISCUSSION THREAD

**Discussion by bestickley** [porsager/postgres#446] [Q&A] [answered]
Hi, I've got a method in a class like so:

```ts
interface CreateUserDto {
  email: string;
  firstName: string;
  lastName: string;
  middleInitial?: string;
  organization: string;
  department: string;
  phone: string;
  altEmail?: string;
  jobTitle: string;
  designation: Designation;
  accessExpirationDate: string;
}
createUser = (user: CreateUserDto) => {
    this.#logger.debug("Inserting into cosmos_user", JSON.stringify(user));
    return this.#sql`
      insert into cosmos_user ${this.#sql(user as Required<CreateUserDto>)}
    `;
  };
```

Unless I cast `CreateUserDto` to a type where all keys are required, I get a TS error, even though I have `undefined` transformed to `null`. I can live with this, but was wondering if there is a better strategy out there? I could do below, but it's more code:

```ts
createUser = (user: CreateUserDto) => {
    this.#logger.debug("Inserting into cosmos_user", JSON.stringify(user));
    const newUser = {altEmail: null, middleInitial: null, ...user };
    return this.#sql`
      insert into cosmos_user ${this.#sql(newUser)}
    `;
  };
```

**Comment by Minigugus** [comment, marked as answer]
I thought `sql` insert helpers calls where already accepting `undefined`... It seams I was wrong :confused: Actually, it's not an issue with the `undefined` transform, but with the insert helper on objects that has some `undefined` properties (it does matter whether or not `undefined` calls are transformed to somethinhg else in this case, as `undefined` properties could just be ignored when encoded anyway).

#447 should fix that :+1:

**Reply by bestickley** [reply]
Wow, thanks so much for the quick response and fix, @Minigugus! Much appreciated.
